### DISCUSSION TITLE
Access denied trying to update a workflow via API

### DISCUSSION THREAD

**Discussion by tronyx** [community/community#24303] [Apps, API and Webhooks]
<p>I created a public repository within my personal account, created a PAT w/ the workflow permissions checked, but am unable to disable the workflow via the API as I receive a message stating that I must have admin rights to the repository. I believe the PAT has the correct permissions so I’m unsure as to why this isn’t working.</p>
<p>Here is the command I am attempting to use (based on the documentation):</p>
<p><code>curl -s christronyxyocum:MY-PAT -X PUT -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/MY-USERNAME/MY-REPO/actions/workflows/workflow.yml/disable</code></p>
<p>I can retrieve information about the workflow without any issues so I believe that I have the correct URL and formatting, etc.</p>

**Comment by tronyx** [comment, marked as answer]
<p>I have figured this out. Rather than using the <code>username:token</code> format like they show repeatedly in their documentation, you must use the <code>-H "Authorization: bearer TOKEN"</code> header with the <code>curl</code> command.</p>

**Comment by francisfuzz** [comment]
<aside class="quote no-group" data-username="christronyxyocum" data-post="2" data-topic="149220">
<div class="title">
<div class="quote-controls"></div>
 christronyxyocum:</div>
<blockquote>
<p>you must use the <code>-H "Authorization: bearer TOKEN"</code> header with the <code>curl</code> command.</p>
</blockquote>
</aside>
<p>For posterity, here’s a friendly link to <a href="https://docs.github.com/en/rest/overview/resources-in-the-rest-api#oauth2-token-sent-in-a-header">those docs for sending an OAuth 2 token in the header</a>. :v:</p>
